cmd/manager/main.go

```diff
@@ -4,140 +4,146 @@ import (
+	"github.com/operator-framework/operator-marketplace/pkg/signals"
 	"github.com/operator-framework/operator-marketplace/pkg/status"
 	sourceCommit "github.com/operator-framework/operator-marketplace/pkg/version"
+
 	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
 	"github.com/operator-framework/operator-sdk/pkg/leader"
 	sdkVersion "github.com/operator-framework/operator-sdk/version"
-	log "github.com/sirupsen/logrus"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
 	kruntime "k8s.io/apimachinery/pkg/runtime"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
 const (
-	initialWait                = time.Duration(1) * time.Minute
-	updateNotificationSendWait = time.Duration(10) * time.Minute
-)
-
-var (
-	version     = flag.Bool("version", false, "displays marketplace source commit info.")
-	tlsKeyPath  = flag.String("tls-key", "", "Path to use for private key (requires tls-cert)")
-	tlsCertPath = flag.String("tls-cert", "", "Path to use for certificate (requires tls-key)")
+	leaderElectionConfigMapName = "marketplace-operator-lock"
 )
 
 func printVersion() {
-	log.Printf("Go Version: %s", runtime.Version())
-	log.Printf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
-	log.Printf("operator-sdk Version: %v", sdkVersion.Version)
+	logrus.Printf("Go Version: %s", runtime.Version())
+	logrus.Printf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
+	logrus.Printf("operator-sdk Version: %v", sdkVersion.Version)
 }
 
 func main() {
 	printVersion()
 
-	// Parse the command line arguments
-	flag.StringVar(&defaults.Dir, "defaultsDir",
-		"", "the directory where the default CatalogSources are stored")
-	var clusterOperatorName string
+	var (
+		clusterOperatorName string
+		tlsKeyPath          string
+		tlsCertPath         string
+		version             bool
+	)
 	flag.StringVar(&clusterOperatorName, "clusterOperatorName", "", "the name of the OpenShift ClusterOperator that should reflect this operator's status, or the empty string to disable ClusterOperator updates")
+	flag.StringVar(&defaults.Dir, "defaultsDir", "", "the directory where the default CatalogSources are stored")
+	flag.BoolVar(&version, "version", false, "displays marketplace source commit info.")
+	flag.StringVar(&tlsKeyPath, "tls-key", "", "Path to use for private key (requires tls-cert)")
+	flag.StringVar(&tlsCertPath, "tls-cert", "", "Path to use for certificate (requires tls-key)")
 	flag.Parse()
 
 	// Check if version flag was set
-	if *version {
-		fmt.Print(sourceCommit.String())
-		// Exit immediately
+	if version {
+		logrus.Infof(sourceCommit.String())
```

Comment:
Probably harmless, but `sourceCommit` isn't supposed to be interpreted as a format string.

```suggestion
		logrus.Info(sourceCommit.String())
```
or

```suggestion
		logrus.Infof("%s", sourceCommit.String())
```